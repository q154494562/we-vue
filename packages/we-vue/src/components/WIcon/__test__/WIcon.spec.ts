import { shallowMount } from '@vue/test-utils'
import Icon from '../'

describe('icon', () => {
  test('create', () => {
    const wrapper = shallowMount(Icon, {
      propsData: {
        type: 'info',
      },
    })

    expect(wrapper.name()).toBe('w-icon')

    expect(wrapper.html()).toMatchSnapshot()

    // is not 'large' by default
    expect(wrapper.vm.large).toBeFalsy()
  })

  test('element class should be computed correctly', () => {
    const wrapper = shallowMount(Icon, {
      propsData: {
        type: 'info',
        large: true,
      },
    })

    // large size icon should has class 'weui-icon_msg'
    expect(wrapper.classes()).toContain('weui-icon_msg')

    const iconClassType = `weui-icon-${wrapper.vm.type}`
    expect(wrapper.classes()).toContain(iconClassType)
  })
})
