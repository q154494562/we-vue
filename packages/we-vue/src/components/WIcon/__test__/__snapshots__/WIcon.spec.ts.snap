// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon create 1`] = `<span class="weui-icon-info"></span>`;
